tests/testthat/test-initial-estimates.R

```diff
@@ -0,0 +1,87 @@
+
+
+describe("initial_estimates", {
+
+  it("matrix_to_df", {
+    initial_est <- get_initial_est(MOD1)
+    omega_df <- matrix_to_df(initial_est$omegas)
+
+    expect_equal(
+      omega_df$init,
+      initial_est$omegas[lower.tri(initial_est$omegas, diag = TRUE)]
+    )
+  })
+
+  it("get_theta_inits", {
+    skip_if_old_nmrec("0.3.0.8001")
+    ctl <- nmrec::read_ctl(get_model_path(MOD1))
+    theta_inits <- get_theta_inits(ctl, mark_flags = "fix")
+
+    expect_equal(
+      names(theta_inits),
+      c("init", "low", "up", "fixed", "record_number")
+    )
+    expect_true(isFALSE(all(theta_inits$fixed)))
+  })
+
+  it("fmt_record_num", {
+    skip_if_old_nmrec("0.3.0.8001")
+    ctl <- nmrec::read_ctl(get_model_path(MOD1))
+
+    # Test THETA
+    theta_inits <- nmrec::extract_theta(ctl)
+    rec_nums <- fmt_record_num(theta_inits)
+    expect_equal(length(rec_nums), length(theta_inits))
+    expect_true(all(rec_nums == 1))
+
+    # Test OMEGA (test multiple cases for added complexity)
+    omega_inits <- nmrec::extract_omega(ctl)
+    rec_nums <- fmt_record_num(omega_inits)
+    expect_equal(length(rec_nums), sum(lower.tri(omega_inits, diag = TRUE)))
+    expect_true(all(rec_nums == 1))
+
+    # Test multiple OMEGA records
+    mod2 <- read_model(file.path(MODEL_DIR_X, "1001"))
+    ctl <- nmrec::read_ctl(get_model_path(mod2))
+    omega_inits <- nmrec::extract_omega(ctl)
+    rec_nums <- fmt_record_num(omega_inits)
+    # Confirm total length
+    expect_equal(length(rec_nums), sum(lower.tri(omega_inits, diag = TRUE)))
+    # Confirm record numbers
+    expect_equal(
+      length(which(rec_nums == 1)),
+      # the first column
+      length(omega_inits[,1])
+    )
+    expect_equal(
+      length(which(rec_nums == 2)),
+      # lower triangular matrix minus the first column
+      sum(lower.tri(omega_inits, diag = TRUE)[,-1])
+    )
+  })
+
+  it("initial_estimates: integration", {
```

Comment:
I would like to have a couple more `initial_estimates()` tests on different models. Mostly to check that the matrices are the expected shape and that the number of specified parameters we parse are what we expect.

Some examples I think are worth testing:

* `inst/model/nonmem/complex/iovmm`
  * usage of `SAME` 
  * multiple `OMEGA` blocks, some of which will show up and some won't (i.e. `record_number` will have `1, 2, 4`)
  * `attr("omega_mat")` is the size we expect it to be
* `inst/model/nonmem/complex/example2_saemimp`
  * expect to see "This model appears to be using... to specify priors" warning
  * check the number of each type (to see that it does indeed parse those priors through)
  * check `record_number` column is what we expect (multiple THETA and OMEGA blocks)

This might be overkill, but there are a lot of moving pieces here and I think it's worth being sure that everything is being passed along the way we want it to.

Reply:
Sorry, my mistake here. I meant to say `"complex/acop-iov"` instead of `"complex/iovmm"` for the first example. Sorry about that.

Using your example code, it looks like this. Notice the `1, 2, 4` in the `record_number` column for OMEGAs (because it skips 3 and 5, which are `SAME` blocks).

```r
> mod_c <- read_model(file.path(MODEL_DIR_X, "acop-iov"))
> initial_est_df <- initial_estimates(mod_c)
> dplyr::select(initial_est_df, parameter_names, init, record_type, record_number)
# A tibble: 10 × 4
   parameter_names  init record_type record_number
   <chr>           <dbl> <chr>               <int>
 1 THETA(1)         2    theta                   1
 2 THETA(2)         3    theta                   1
 3 THETA(3)        10    theta                   1
 4 THETA(4)         0.02 theta                   1
 5 THETA(5)         1    theta                   1
 6 OMEGA(1,1)       0.05 omega                   1
 7 OMEGA(2,2)       0.2  omega                   1
 8 OMEGA(3,3)       0.2  omega                   2
 9 OMEGA(33,33)     0.2  omega                   4
10 SIGMA(1,1)       1    sigma                   1
```